Today's commitment creation flow's bug fixes

jkhabra opened the pull request.

jkhabra committed:
refactor: create popup component

jkhabra committed:
feat: add estimation drop down in add-daily-task form

bitspook commented on ui/src/components/Form/SelectBox/index.tsx:
```diff
@@ -22,27 +22,28 @@ interface FileFieldProps {
   options: SelectFieldItem[];
   isLoading?: boolean;
   onSelect?: any;
```
We don't need to make this type any
```suggestion
  onSelect: (item: SelectFieldItem) => void;
```

bitspook commented on ui/src/components/Form/SelectBox/index.tsx:
```diff
@@ -22,27 +22,28 @@ interface FileFieldProps {
   isLoading?: boolean;
   onSelect?: any;
-  Header?: React.FC;
+  FirstValue?: React.FC;
```
Let's convert this to 2 props: `value` for whatever is the selected value of the component, and `Selected`, which is the component which will get rendered with the selected value. 
```Selected: React.FC<{ item: SelectFieldItem }>```

bitspook commented on ui/src/components/Modal/Popup/index.tsx:
```diff
@@ -0,0 +1,39 @@
+
+interface Props {
+  onClose: () => void;
+  setIsDirtyPopupOpen?: (isOpen: boolean) => void;
```
Let's turn these to: `onAccept` and `onCancel`.

bitspook commented on ui/src/components/SlideBar/index.tsx:
```diff
@@ -20,30 +20,17 @@ const SlideBar: React.FC<Props> = (p) => {
     <Modal onClose={() => setIsDirtyPopupOpen(false)} isOpen={Boolean(p.isDirtyPopupOpen)}>
-      <Popup />
+      <Popup
+        title={'Are you sure'}
```
Let's turn Popup into a variation of Modal instead of just being its content.

jkhabra committed:
refactor: change popup name into warning component

- rap up warning component in modal component
- improve daily status row styles
- refactor SelectBox component
